src/rust/engine/src/context.rs

```diff
@@ -222,6 +222,53 @@ impl Core {
   pub fn http_client(&self) -> reqwest::r#async::Client {
     self.store_and_command_runner_and_http_client.get().2
   }
+
+  ///
+  /// Start running a Future on a tokio Runtime.
+  ///
+  pub fn spawn<F: Future<Item = (), Error = ()> + Send + 'static>(&self, future: F) {
+    // Make sure to copy our (thread-local) logging destination into the task.
+    // When a daemon thread kicks off a future, it should log like a daemon thread (and similarly
+    // for a user-facing thread).
+    let logging_destination = logging::get_destination();
+    self
+      .runtime
+      .get()
+      .read()
+      .executor()
+      .spawn(futures::future::ok(()).and_then(move |()| {
+        logging::set_destination(logging_destination);
+        future
+      }))
+  }
+
+  ///
+  /// Run a Future and return its resolved Result.
+  ///
+  /// This should never be called from in a Future context, or any context where anyone may want to
```

Comment:
Not worth blocking merge on probably, but if there are upstream docs to link on this topic, that would be handy.